staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/compilation.go

```diff
@@ -141,7 +141,10 @@ func compileRule(rule apiextensions.ValidationRule, env *cel.Env) (compilationRe
 		}
 	}
 
-	prog, err := env.Program(ast, cel.EvalOptions(cel.OptOptimize))
+	prog, err := env.Program(ast,
+		cel.EvalOptions(cel.OptOptimize),
+		cel.OptimizeRegex(library.ExtensionLibRegexOptimizations...),
```

Comment:
Do we want to add CostTracking(CostEstimator) for extension func cost cal later?

Reply:
Yes, there reason you're not seeing it here is because I want to integrate with https://github.com/kubernetes/kubernetes/pull/108419 after it merges. I should have made this clear in the description. I'll add an explanation now.

Reply:
This is all integrated now.